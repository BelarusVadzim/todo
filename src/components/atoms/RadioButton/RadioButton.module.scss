.radioButton{
  /* HIDE RADIO */
  [type=radio] { 
    position: absolute;
    opacity: 0;
    width: 0;
    height: 0;
  }

  /* NORMAL STYLES */
  [type=radio] + span {
    cursor: pointer;
    color: rgb(158, 156, 156);
    font-size: 14px;
  }

  [type=radio] + span:hover {
    color: rgb(80, 80, 80);
  }

  /* CHECKED STYLES */
  [type=radio]:checked + span {
    color: rgb(66, 175, 226);
  }
}
