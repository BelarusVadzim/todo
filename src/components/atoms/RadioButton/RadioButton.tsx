import { PropsWithChildren } from "react";
import style from "./RadioButton.module.scss";
import { PropsWithClassName } from "types";

export type RadioButtonProps = PropsWithChildren &
  PropsWithClassName & {
    groupName: string;
    selectedValue?: string;
    onChange: (value: string) => void;
    id?: string;
    value: string;
  };

const RadioButton: React.FC<RadioButtonProps> = ({
  children,
  groupName,
  selectedValue,
  onChange,
  id,
  value,
  className = "",
}) => {
  let combinedClassName = `${style.radioButton} ${className}`;

  return (
    <label className={combinedClassName} htmlFor={id}>
      <input
        id={id}
        type="radio"
        name={groupName}
        value={value as string}
        checked={value === selectedValue}
        onChange={(val) => onChange(val.target.value)}
      />
      <span>{children}</span>
    </label>
  );
};

export default RadioButton;
