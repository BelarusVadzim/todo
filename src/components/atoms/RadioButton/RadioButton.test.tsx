import render  from 'react-test-renderer';
import RadioButton from './RadioButton';

jest.mock('./RadioButton.module.scss', () => ({ labelLineThrough: 'labelLineThroughstyle', label: 'labelStyle' }));
const groupName = 'groupName';
const value = 'value';


describe('<Label />', () => {
  it('should render properly', () => {
    const component = render
      .create(<RadioButton className='className' groupName={groupName} value={value}>
        LabelChild
      </RadioButton>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('when onChange raised and onChange handler specified should call expected functions with expected values', () => {
    const onChangeHandler = jest.fn();
    const component = render
      .create(<RadioButton 
        onChange={onChangeHandler}
        groupName={groupName} 
        value={value}>radioButtonText</RadioButton>);

    const expectedValue = value;
    const changeEventArg = { target: { value: expectedValue } };

    component.root.findByType('input').props.onChange(changeEventArg);

    expect(onChangeHandler).toBeCalledTimes(1);
    expect(onChangeHandler).toBeCalledWith(expectedValue);
  });
});
