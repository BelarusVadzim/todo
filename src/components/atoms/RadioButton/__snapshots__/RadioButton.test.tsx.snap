// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render properly 1`] = `
<label
  className="undefined className"
>
  <input
    checked={false}
    name="groupName"
    onChange={[Function]}
    type="radio"
    value="value"
  />
  <span>
    LabelChild
  </span>
</label>
`;
